pkg/clusteragent/autoscaling/workload/provider.go

```diff
@@ -0,0 +1,50 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build kubeapiserver
+
+package workload
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/DataDog/datadog-agent/pkg/clusteragent/autoscaling"
+	"github.com/DataDog/datadog-agent/pkg/clusteragent/autoscaling/workload/model"
+	"github.com/DataDog/datadog-agent/pkg/util/kubernetes/apiserver"
+	"github.com/DataDog/datadog-agent/pkg/util/kubernetes/apiserver/leaderelection"
+)
+
+// StartWorkloadAutoscaling starts the workload autoscaling controller
+func StartWorkloadAutoscaling(ctx context.Context, apiCl *apiserver.APIClient, rcClient rcClient) error {
+	if apiCl == nil {
+		return fmt.Errorf("Impossible to start workload autoscaling without valid APIClient")
+	}
+
+	le, err := leaderelection.GetLeaderEngine()
+	if err != nil {
+		return fmt.Errorf("Unable to start workload autoscaling as LeaderElection failed with: %v", err)
+	}
+
+	store := autoscaling.NewStore[model.PodAutoscalerInternal]()
```

Comment:
Is it worth using `autoscaling.NewStore[*model.PodAutoscalerInternal]` ? `model.PodAutoscalerInternal` is a large struct